chainerx_cc/chainerx/routines/math.cc

```diff
@@ -1176,4 +1176,28 @@ Array IsFinite(const Array& x) {
     return out;
 }
 
+Array Where(const Array& condition, const Array& x, const Array& y) {
+    if (condition.dtype() != Dtype::kBool) {
```

Comment:
How about casting `condition` to bool dtype if the given condition is not a bool array?

Reply:
Done. However do note that as per 
https://github.com/onnx/onnx/blob/master/docs/Operators.md#Where

`condition` should be `bool`

Reply:
For consistency with logical functions (e.g., `logical_not`), the condition should allow all dtypes.

Reply:
Oh. Have done it in the recent commits.